src/Planners/include/BipedalLocomotion/Planners/ContactList.h

```diff
@@ -0,0 +1,217 @@
+/**
+ * @file ContactList.h
+ * @authors Stefano Dafarra
+ * @copyright 2020 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#ifndef BIPEDAL_LOCOMOTION_PLANNERS_CONTACTLIST_H
+#define BIPEDAL_LOCOMOTION_PLANNERS_CONTACTLIST_H
+
+// BipedalLocomotion
+#include <BipedalLocomotion/Planners/Contact.h>
+
+//iDynTree
+#include <iDynTree/Core/Transform.h>
+
+//std
+#include <set>
+#include <string>
+#include <functional>
+
+namespace BipedalLocomotion
+{
+namespace Planners
+{
+
+/**
+ * @brief Class containing a list of contacts.
+ * The contact are added such that the activation time is strictly growing. In addition, contacts cannot be overlapping.
+ * It represents a series of contact activations and deactivations of a single entity.
+ */
+class ContactList
+{
+    /**
+     * @brief Struct used for inserting new contacts in the set.
+     */
+    struct ContactCompare {
+        bool operator()(const Contact& lhs, const Contact& rhs) const;
+    };
+
+    std::set<Contact, ContactCompare> m_contacts; /** Data structure for inserting and ordering contacts. **/
+    std::string m_defaultName{"ContactList"}; /** Default name for the contact list. **/
+    ContactType m_defaultContactType{ContactType::FULL}; /** Default contact type. **/
+
+public:
+
+    using const_iterator = std::set<Contact, ContactCompare>::const_iterator;
+    using const_reverse_iterator = std::set<Contact, ContactCompare>::const_reverse_iterator;
+
+    /**
+     * @brief Set the default name.
+     * @param defaultName the default name.
+     */
+    void setDefaultName(const std::string& defaultName);
```

Comment:
Actually, I would avoid this to prevent stupid mistakes like
```c++
if (list.defaultName() = "a")
```
In addition, even if it is not the case now, in the future I may need to do some changes when this name or the contact type change.